### DISCUSSION TITLE
Add a Ctrl+P palate to switch between tables, files, etc. like in VS Code

### DISCUSSION THREAD

**Discussion by DeflateAwning** [dbeaver/dbeaver#19538] [General]
**Is your feature request related to a problem? Please describe.**
I am always frustrated when I have to switch between tables in databases with hundreds of tables. Scrolling through the list is annoying.

It's even worse when there are multiple databases open, and even worse if there are multiple databases with the same schema (like a staging and a production database).

**Describe the solution you'd like**
I want to be able to press Ctrl+P, and then a command palate pops up (like in VS Code) which allows me to select a table or a file by typing a substring of the table name (with fuzzy matching, probably).

**Describe alternatives you've considered**
The only way to solve this problem right now is to scroll through the list. You can jump to a letter by pressing a letter, but that doesn't always reliably jump within the right database, it seems.

**Additional context**
Here's an example of what the feature looks like in VS Code: 
![image](https://user-images.githubusercontent.com/11021263/227311068-084637fe-0499-40b5-851b-2d589df8ed18.png)

It obviously only searches files, but ideally the one in DBeaver would search both files and tables (and maybe even users, schemas, everything in the side panel, etc.).

**Comment by ShadelessFox** [comment]
Hello @DeflateAwning,

You can quickly switch between editors by pressing <kbd>CTRL+E</kbd> (fuzzy search is available there):

![image](https://user-images.githubusercontent.com/35821147/227408751-92a020b0-0b2f-4a00-a1a1-ef8c79446d3e.png)

Or by clicking on the chevron that is shown when editor tabs can't fit in a single line:

![image](https://user-images.githubusercontent.com/35821147/227409160-93ddc258-4189-4249-9f4c-bbe60b956ac1.png)

Please let me know if it solves your issue.

**Comment by DeflateAwning** [comment]
Ctrl+E only switches between already-open editors/tabs. It would be great if it could open a new table from the list in the far-left "Database Navigator" pane. Not sure if adding to Ctrl+E is the way to go, or if a new Ctrl+P option (similar to VSCode) would be better

**Comment by ShadelessFox** [comment]
I see. For that case, you can use <kbd>Ctrl+Shift+D</kbd>:

![image](https://user-images.githubusercontent.com/35821147/228090976-09488f6c-1f7b-4f58-8fed-3e07ef666578.png)

Unfortunately, fuzzy search is not available there since results are filtered on the database side.

**Comment by DeflateAwning** [comment]
This is awesome! What inspired the Ctrl+Shift+D shortcut for it? Most programs use Ctrl+P for this sort of functionality

**Reply by nacesprin** [reply]
Menu: Window > Preferences > User Interface > Keyboard.
Remove link Ctrl+P (for printer) and set it for "Open Database Object"

**Comment by ShadelessFox** [comment]
CTRL+P is already in use by Eclipse, which might be the reason. Also, it's easy to execute CTRL+SHIFT+D using one hand, whereas CTRL+P would require two hands, which is a con IMO.

If you want to change keyboard shortcuts, you can do that in the preferences -> User Interface -> Keys.

**Comment by DeflateAwning** [comment]
Awesome, thank you!

**Comment by DeflateAwning** [comment]
Ctrl+Shift+D is totally the solution. I personally re-bound it to Ctrl+P. Would be nice if it could be bound to multiple hotkeys. I feel like this feature as a well-known default is far more useful than Ctrl+P being available in 2024, but I can live with this for now.
